File: QTD.SPA/src/app/_DtoModels/StudentEvaluation/RatingScaleN.ts
import { Entity } from "../Entity";

export class RatingScaleN extends Entity{
  ratingScaleDescription!:string;
  ratingScaleExpanded:RatingScaleExpanded[]= [];
}

export class RatingScaleExpanded extends Entity{
  description!:string;
  ratings!:number;
}

File: QTD2.Domain/Entities/Core/RatingScaleN.cs
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using QTD2.Domain.Entities.Common;

namespace QTD2.Domain.Entities.Core
{
    public class RatingScaleN : Entity
    {
        public string RatingScaleDescription { get; set; }

        public virtual ICollection<StudentEvaluation> StudentEvaluations { get; set; } = new List<StudentEvaluation>();
        public virtual ICollection<RatingScaleExpanded> RatingScaleExpanded { get; set; } = new List<RatingScaleExpanded>();

        public RatingScaleN()
        {

        }
        public RatingScaleN(string description)
        {
            RatingScaleDescription = description;
        }
    }
    
}
